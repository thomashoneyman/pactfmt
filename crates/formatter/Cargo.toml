[package]
name = "formatter"
version = "0.1.0"
edition = "2021"

[dependencies]
syntax = { path = "../syntax" }
pretty = "0.12.3"

[dev-dependencies]
insta = "1.41.1"
pretty_assertions = "1.4.0"
test_each_file = "0.3.4"
test-utils = { path = "../test-utils" }
