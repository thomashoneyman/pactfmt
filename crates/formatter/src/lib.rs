mod format_doc;
mod format_tree;
mod lower_tree;

use pretty::RcAllocator;
use syntax::parser::parse;
use syntax::tokenize;

use crate::format_doc::FormatDoc;

pub fn format_source(input: &str, width: usize) -> String {
    let tokens = tokenize(input);
    let parsed = parse(tokens);
    let lowered = lower_tree::lower_file(parsed);
    let format_doc = lowered
        .iter()
        .fold(FormatDoc::nil(&RcAllocator), |acc, fst| {
            acc.append(fst.format(&RcAllocator))
        });
    format_doc.pretty(width)
}

pub fn check_source(input: &str) -> bool {
    let tokens = tokenize(input);
    let parsed = parse(tokens);
    parsed.kind == syntax::types::TreeKind::File
}
