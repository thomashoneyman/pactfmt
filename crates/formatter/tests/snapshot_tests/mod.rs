use formatter::format_source;
use test_each_file::test_each_file;

test_each_file! { in "crates/formatter/tests/snapshot_tests/fixtures" => |content: &str| {
    insta::assert_snapshot!(format_source(content, 60).unwrap());
}}
