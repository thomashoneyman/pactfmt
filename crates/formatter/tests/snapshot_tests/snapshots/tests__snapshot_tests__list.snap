---
source: crates/formatter/tests/snapshot_tests/mod.rs
expression: "format_source(content, 60)"
---
[
  1,
  2, ; trailing comment
  ; leading comment
  3
]
